binders/kafka-binder/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -619,6 +628,11 @@ protected MessageProducer createConsumerEndpoint(
 								? new ContainerProperties(Pattern.compile(topics[0]))
 								: new ContainerProperties(topics)
 						: new ContainerProperties(topicPartitionOffsets);
+
+		if (this.configurationProperties.isEnableObservation() || extendedConsumerProperties.getExtension().isEnableObservation()) {
```

Comment:
I guess it's possible for them to enable it at the binder level and then disable it at a specific binding level.

OTOH, I wonder if we need binding level properties at all - they could always configure multiple binders if they want that level of control.